src/components/VaultItemTiles.jsx

```diff
@@ -58,27 +63,27 @@ export default function VaultItemTiles({ items }) {
       const vaultItem = itemsArray[vaultItemIndex];
       const deleteClicked = event.target.closest("button")
       if (deleteClicked) {
-        await handleDeleteVaultItem(vaultItem.uuid, vaultItemIndex);
+        await handleDeleteVaultItem(vaultItem, vaultItemIndex);
       } else {
         openPopOut(event, vaultItem.data);
       }
     }
   };
 
-  const handleDeleteVaultItem = async (uuid, vaultItemIndex) => {
+  const loadingStates = (vaultItemIndex, isLoading) => {
```

Comment:
Nice refactoring!

Reply:
Hahaha thanks! I was looking at the code today and thought, why did I copy/paste this 3 times lol